src/ui/public/vislib/lib/dispatch.js

```diff
@@ -161,7 +161,10 @@ define(function (require) {
      */
     Dispatch.prototype.allowBrushing = function () {
       var xAxis = this.handler.xAxis;
-      return Boolean(xAxis.ordered && xAxis.xScale && _.isFunction(xAxis.xScale.invert));
+      // Don't allow brushing for time based charts from non-time-based indices
+      var hasTimeField = this.handler.vis._attr.hasTimeField;
+
+      return Boolean(xAxis.ordered && xAxis.xScale && _.isFunction(xAxis.xScale.invert) && hasTimeField);
```

Comment:
Nitpick, but putting `hasTimeField` first would allow the value to resolve false faster when not using time-based events. 

Otherwise, I like this solution a lot better!